src/main/java/io/aiven/kafka/connect/http/config/HttpSinkConfig.java

```diff
@@ -287,11 +394,76 @@ public final String connectorName() {
         return originalsStrings().get(NAME_CONFIG);
     }
 
+    public final URI oauth2AccessTokenUri() {
+        return getString(OAUTH2_ACCESS_TOKEN_URL_CONFIG) != null ? toURI(OAUTH2_ACCESS_TOKEN_URL_CONFIG) : null;
+    }
+
+    private URI toURI(final String propertyName) {
+        try {
+            return new URL(getString(propertyName)).toURI();
+        } catch (final MalformedURLException | URISyntaxException e) {
```

Comment:
Could it actually happend? I think we checked this value with URL Validator?

Reply:
well it cannot, but `MalformedURLException` and `URISyntaxException` are checked exceptions. `URL` and `URI` constructors throw theme explicitly.